src/Engine/GraphLayout/GraphLayout.cs

```diff
@@ -256,18 +323,29 @@ public void AssignLayers()
                     layerWidth = 0;
 
                     if (n.LeftEdges.Count == 0)
+                    {
                         isFinalLayer = true;
+                    }
                 }
 
                 AddToLayer(n, currentLayer);
                 processed++;
 
                 if (n.Width > layerWidth)
+                {
                     layerWidth = n.Width;
+                }
             }
 
-            // Put the input nodes on the leftmost layer
-            AddToLayer(Nodes.Where(x => x.LeftEdges.Count == 0).ToList(), Layers.Count);
+            // Put all input nodes and isolated nodes on the leftmost layer
+            AddToLayer(Nodes.Where(x =>
+                (x.LeftEdges.Count == 0 && !x.HasUnconnectedInPort) ||
```

Comment:
@aparajit-pratap This line of code refers to the nodes without any input ports at all.

If there is a node that has only one empty input port in it, that node will not have any `LeftEdges` but `HasUnconnectedInPort` is true, therefore it should not be put on the leftmost layer.